src/toolkit/res_buf.h

```diff
@@ -101,16 +101,15 @@ class ResBuf {
   /// Returns true if there are no resources in the buffer.
   inline bool empty() const { return rs_.empty(); }
 
-  /// Pops and returns the specified quantity from the buffer as a single
-  /// resource object.
+  /// Pops and returns the specified quantity from the buffer as vector of 
+  /// resources.
   /// Resources are split if necessary in order to pop the exact quantity
   /// requested (within eps_rsrc()).  Resources are retrieved in the order they
-  /// were pushed (i.e. oldest first) and are squashed into a single object
-  /// when returned.
+  /// were pushed (i.e. oldest first).
   ///
   /// @throws ValueError the specified pop quantity is larger than the
   /// buffer's current inventory.
-  typename T::Ptr Pop(double qty) {
+  std::vector<typename T::Ptr> PopVector(double qty) {
```

Comment:
Do you think it would confusing to name this `PopN(double qty)`.  It behaves a lot like `PopN` except it chooses how many items by `qty` instead of `count`?

Reply:
A clearer and less confusing name could be something like `PopByQuantity(double qty)` or `PopAmount(double qty)` to make it clear that it operates based on a quantity rather than a count.

Reply:
@gonuke I personally think re-using `PopN` with a quantity would be confusing. 

@ahnaf-tahmid-chowdhury I also don't want it to confuse people with regular `Pop`, which is in heavy use already. I expect that will continue to be the most frequently used way to pop off of a ResBuf. To me, `PopByQuantity` or `PopAmount` doesn't distinguish it from `Pop`, it actually makes it more likely that people get confused. While the code is now written that this function is the workhorse of popping, in my mind it's a sub-type of Pop, just pop without squash.

Reply:
`PopWithoutSquash`/`PopNonSquashed`/`SimplePop`/`BasicPop`? Just some suggestions.

Ultimately, the goal is to choose a name that accurately describes the function's behavior while minimizing confusion with existing methods like Pop.